src/index.js

```diff
@@ -73,9 +73,16 @@ function renderToString(vnode, context, opts, inner, isSvgMode) {
 				c._disable = c.__x = true;
 				c.props = props;
 				c.context = context;
-				if (nodeName.getDerivedStateFromProps) c.state = assign(assign({}, c.state), nodeName.getDerivedStateFromProps(c.props, c.state));
-				else if (c.componentWillMount) c.componentWillMount();
-				rendered = c.render(c.props, c.state, c.context);
+				try {
+					if (nodeName.getDerivedStateFromProps) c.state = assign(assign({}, c.state), nodeName.getDerivedStateFromProps(c.props, c.state));
+					else if (c.componentWillMount) c.componentWillMount();
+					rendered = c.render(c.props, c.state, c.context);
```

Comment:
You likely will want to wrap the recursive call to `renderString` in try/catch instead of the call to the component's `render` method so only errors of children will be caught.

Reply:
yea, this was a correct observation, it has been fixed